queue/main_test.go

```
@@ -78,29 +109,39 @@ func TestGetQueueInstance(t *testing.T) {
-		sessionWrapper.Channel = &amqp.Channel{}
+	t.Run("CreateChannel should not throw error from ChannelFn() method", func(t *testing.T) {
+		mockQueue := &QueueWrapper{ChannelFn: func() (*amqp.Channel, error) {
+			return &amqp.Channel{}, nil
+		}}
+		assert.NoError(t, mockQueue.CreateChannel())
+	})
+	t.Run("QueueWrapper should always implement DeclareQueue() method", func(t *testing.T) {
+		queueWrapper.Channel = &amqp.Channel{}
 		assert.Panics(t, func() {
-			sessionWrapper.declareQueue()
+			queueWrapper.DeclareQueue()
```

golangci-lint (1.64.8)

Error return value of `queueWrapper.DeclareQueue` is not checked

(errcheck)